From 10371ea77a78d9cee13b5adfda8b855432bf2647 Mon Sep 17 00:00:00 2001
From: Samuli Piippo <samuli.piippo@qt.io>
Date: Tue, 21 Jan 2020 12:45:38 +0200
Subject: [PATCH 2/4] qtbase: use syncqt.pl from QT_HOST_PATH

Use syncqt.pl from QT_HOST_PATH if that is given, since qtbase
sources might not be available and CMAKE_INSTALL_PREFIX doesn't
check for sysroot.

Change-Id: I165b17a5a02fd4dbb2340bf69a641b8aaab8fabd
---
 cmake/QtBuild.cmake | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/cmake/QtBuild.cmake b/cmake/QtBuild.cmake
index 738f38fbd1..e4bb6f0e40 100644
--- a/cmake/QtBuild.cmake
+++ b/cmake/QtBuild.cmake
@@ -777,6 +777,12 @@ function(qt_ensure_sync_qt)
         qt_path_join(syncqt_install_dir ${QT_INSTALL_DIR} ${INSTALL_LIBEXECDIR})
         qt_copy_or_install(PROGRAMS "${SYNCQT_FROM_SOURCE}"
                            DESTINATION "${syncqt_install_dir}")
+    elseif(QT_HOST_PATH)
+        get_filename_component(syncqt_absolute_path
+                               "${QT_HOST_PATH}/${INSTALL_LIBEXECDIR}/syncqt.pl"
+                               ABSOLUTE)
+        set(QT_SYNCQT "${syncqt_absolute_path}" CACHE FILEPATH "syncqt script")
+        message(STATUS "Using host syncqt found at: ${QT_SYNCQT}")
     else()
         get_filename_component(syncqt_absolute_path
                                "${CMAKE_INSTALL_PREFIX}/${INSTALL_LIBEXECDIR}/syncqt.pl"
